ntbs-service/Pages/Notifications/Overview.cshtml.cs

```diff
@@ -69,13 +69,22 @@ public async Task<IActionResult> OnGetAsync()
             }
 
             CultureAndResistance = await _cultureAndResistanceService.GetCultureAndResistanceDetailsAsync(NotificationId);
-            TreatmentPeriods = Notification.TreatmentEvents.GroupEpisodesIntoPeriods();
+
+            var treatmentEvents = FilterEventsForPostMortem();
+            TreatmentPeriods = treatmentEvents.GroupEpisodesIntoPeriods();
 
             AddDenotificationEventIfDenotified();
             CalculateTreatmentOutcomes();
             return Page();
         }
 
+        private IEnumerable<TreatmentEvent> FilterEventsForPostMortem()
+        {
+            return Notification.IsPostMortemAndHasCorrectEvents
+                ? Notification.TreatmentEvents.Where(te => te.TreatmentEventTypeIsOutcome && te.TreatmentOutcome.TreatmentOutcomeType == TreatmentOutcomeType.Died)

```

Comment:
This isn't correct - we do want to show both events, it's just that we want to show them in a one day period (so the diagnosis date will fall outside of the period it appears in). It's a bit weird, but we thought it best because we do want to show the diagnosis, but it would be strange for the period to include dates after the patient died.